drivers/src/pci/root.rs

```diff
@@ -424,19 +359,77 @@ impl PcieRoot {
         if !valid_config_mmio_access(offset, len) {
             return Err(Error::UnsupportedConfigAccess);
         }
-        let mut devices = self.devices.lock();
-        let (dev_id, dev_offset) = devices.virtual_config_offset_to_device(offset as usize)?;
+        let (dev_id, dev_offset) = self.virtual_config_offset_to_device(offset as usize)?;
         // If the device ID is present in the hierarchy, then it must be in the arena.
-        let dev = devices.device_arena.get_mut(dev_id).unwrap();
+        let dev = self.device_arena.get(dev_id).unwrap();
```

Comment:
Unwrap might need comment unless obvious from context.

Reply:
There's a comment already on the line right above this.